src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -86,6 +87,29 @@ internal class InProcOrRemoteHostAnalyzerRunner
             }
         }
 
+        public static async Task<ImmutableArray<Diagnostic>> GetSourceGeneratorDiagnosticsAsync(Project project, CancellationToken cancellationToken)
+        {
+            var options = project.Solution.Services.GetRequiredService<IWorkspaceConfigurationService>().Options;
+            if (!options.RunSourceGeneratorsInSameProcessOnly)
+            {
+                var remoteHostClient = await RemoteHostClient.TryGetClientAsync(project, cancellationToken).ConfigureAwait(false);
+                if (remoteHostClient != null)
+                {
+                    var result = await remoteHostClient.TryInvokeAsync<IRemoteDiagnosticAnalyzerService, ImmutableArray<DiagnosticData>>(
+                        project.Solution,
+                        invocation: (service, solutionInfo, cancellationToken) => service.GetSourceGeneratorDiagnosticsAsync(solutionInfo, project.Id, cancellationToken),
+                        cancellationToken).ConfigureAwait(false);
+                    
+                    if (!result.HasValue || result.Value.IsEmpty)
```

Comment:
don't need the .IsEmpty check here.  the .ToDiagnosticsAsync will do the right thing.